src/main/java/ca/corefacility/bioinformatics/irida/ria/web/projects/metadata/ProjectSampleMetadataController.java

```diff
@@ -123,26 +124,19 @@ public SampleMetadataStorage createProjectSampleMetadata(HttpSession session, @P
 		if (stored != null) {
 			stored.setSampleNameColumn(sampleNameColumn);
 			Project project = projectService.read(projectId);
-			List<Map<String, String>> rows = stored.getRows();
-
```

Comment:
Sorry don't know where to exactly leave this comment.  Can you move this functionality to an ajax controller.  I know this is how it was done, but we are really trying to get rid of returning `Map<String, Object>`, In the ajax controller you can set the response type to `ResponseEntity` and set the status on it if you need to, I actually don't think you need to return anything from this though, if there is no error thrown, it will return success.  Most of the contents of the method should also be in your service (right?).

Reply:
ProjectSampleMetadataAjaxController.java was created in the last PR. I left ProjectSampleMetadataController.java, because I was trying to implement the new uploader beside the old uploader. My plan was to create a seperate PR to remove unused old code. I also want to create a seperate PR after the review page is done for tests.